syntax = "proto3";

package message;

service VerifyService {
  rpc GetVerifyCode (GetVerifyReq) returns (GetVerifyRsp) {}
}

message GetVerifyReq {
  string email = 1;
}

message GetVerifyRsp {
  int32 error = 1;
  string email = 2;
  string code = 3;
}

message GetChatServerReq {
  int32 uid = 1;
}

message GetChatServerRsp {
  int32 error = 1;
  string host = 2;
  string port = 3;
  string token = 4;
}

message LoginReq{
	int32 uid = 1;
	string token = 2;
}

message LoginRsp{
	int32 error = 1;
	int32 uid = 2;
	string token = 3;
}

service StatusService {
	rpc GetChatServer (GetChatServerReq) returns (GetChatServerRsp) {}
	rpc Login(LoginReq) returns(LoginRsp);
}

message AddFriendReq{
	int32 applyuid = 1;
	string name = 2;
	string desc = 3;
	int32 touid = 4;
}

message AddFriendRsp{
	int32 error = 1;
	int32 applyuid = 2;
	int32 touid = 3;
}

message ReplyFriendReq{
	int32 replyuid = 1;
	bool agree = 2;
	int32 touid = 3;
}

message ReplyFriendRsp{
	int32 error = 1;
	int32 replyuid = 2;
	int32 touid = 3;
}

message SendChatMsgReq{
	int32 fromuid = 1;
	int32 touid = 2;
	string message = 3;
}

message SendChatMsgRsp{
	int32 error = 1;
	int32 fromuid = 2;
	int32 touid = 3;
}

message AuthFriendReq{
	int32 fromuid = 1;
	int32 touid = 2;
}

message AuthFriendRsp{
	int32 error = 1;
	int32 fromuid = 2;
	int32 touid = 3;
}

message TextChatMsgReq{
	int32 fromuid = 1;
	int32 touid = 2;
	repeated TextChatData textmsgs = 3;
}

message TextChatData{
	string msgid = 1;
	string msgcontent = 2;
}

message TextChatMsgRsp{
	int32 error = 1;
	int32 fromuid = 2;
	int32 touid = 3;
	repeated TextChatData textmsgs = 4;
}

// 好友查询请求
message SearchFriendReq{
	int32 uid = 1;
	string keyword = 2;  // 搜索关键词（用户名或邮箱）
}

// 好友查询响应
message SearchFriendRsp{
	int32 error = 1;
	repeated UserInfo users = 2;
}

// 用户信息
message UserInfo{
	int32 uid = 1;
	string name = 2;
	string email = 3;
	string nick = 4;
	string icon = 5;
	int32 sex = 6;
	string desc = 7;
}

// 获取好友申请列表请求
message GetFriendRequestsReq{
	int32 uid = 1;
}

// 获取好友申请列表响应
message GetFriendRequestsRsp{
	int32 error = 1;
	repeated ApplyInfo requests = 2;
}

// 申请信息
message ApplyInfo{
	int32 uid = 1;
	string name = 2;
	string desc = 3;
	string icon = 4;
	string nick = 5;
	int32 sex = 6;
	int32 status = 7;  // 0:待处理 1:已同意 2:已拒绝
}

// 获取我的好友列表请求
message GetMyFriendsReq{
	int32 uid = 1;
}

// 获取我的好友列表响应
message GetMyFriendsRsp{
	int32 error = 1;
	repeated UserInfo friends = 2;
}

service ChatService{
	rpc NotifyAddFriend(AddFriendReq) returns(AddFriendRsp){}
	rpc ReplyAddFriend(ReplyFriendReq) returns(ReplyFriendRsp){}
	rpc SendChatMsg(SendChatMsgReq) returns(SendChatMsgRsp){}
	rpc NotifyAuthFriend(AuthFriendReq) returns(AuthFriendRsp){}
	rpc NotifyTextChatMsg(TextChatMsgReq) returns(TextChatMsgRsp){}
	rpc SearchFriend(SearchFriendReq) returns(SearchFriendRsp){}
	rpc GetFriendRequests(GetFriendRequestsReq) returns(GetFriendRequestsRsp){}
	rpc GetMyFriends(GetMyFriendsReq) returns(GetMyFriendsRsp){}
}